python/tk_framework_desktopserver/certificates.py

```diff
@@ -46,12 +48,41 @@ def exists(self):
         """
         return os.path.exists(self._cert_path) and os.path.exists(self._key_path)
 
-    def remove_files(self):
+    def backup_files(self):
         """
-        Removes the files from the
+        Backups the certificate files to the backup/year-month-day-hour-minute-second folder.
         """
-        os.unlink(self._cert_path)
-        os.unlink(self._key_path)
+        parent_folder = os.path.dirname(self._cert_path)
+        backup_folder = os.path.join(
+            parent_folder,
+            "backups",
+            datetime.datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+        )
+        self._backup_file(self._cert_path, backup_folder)
+        self._backup_file(self._key_path, backup_folder)
+
+    def _backup_file(self, src_file, dst_folder):
+        """
+        Takes a certificate file (if it exists) and moves it to the destination folder.
+
+        :param src_file: File to move.
+        :param dst_folder: Folder in which we will be backing up the file.
+        """
+
+        # If the certificate file does not exist, no need to backup.
+        if not os.path.exists(src_file):
+            return
+
+        # If the backup folder does not exist, create it.
+        if not os.path.exists(dst_folder):
+            os.makedirs(dst_folder)
```

Comment:
better to use https://github.com/shotgunsoftware/tk-core/blob/master/python/tank/util/filesystem.py#L115 to ensure we write with open permissions?